integration/kubernetes/confidential/snp.bats

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bats
+# Copyright 2023 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Environment variables
+TEST_TAG="[cc][kubernetes][containerd][snp]"
+TESTS_REPO_DIR=$(realpath "${BATS_TEST_DIRNAME}/../../..")
+RUNTIMECLASS="${RUNTIMECLASS:-"kata"}"
+IMAGE_REPO="ghcr.io/confidential-containers/test-container"
+UNENCRYPTED_IMAGE_URL="${IMAGE_REPO}:unencrypted"
+
+# Text to grep for active feature in guest dmesg output
+SNP_DMESG_GREP_TEXT="Memory Encryption Features active:.*SEV-SNP"
+
+export TEST_DIR
+export ENCRYPTION_KEY
+export SSH_KEY_FILE
+
+load "${BATS_TEST_DIRNAME}/../../confidential/lib.sh"
+load "${TESTS_REPO_DIR}/lib/common.bash"
+
+
+# Delete all test services
+k8s_delete_all() {
+	k8s_delete_by_yaml "snp-unencrypted" "${TEST_DIR}/snp-unencrypted.yaml"
+}
+
+setup_file() {
+  TEST_DIR="$(mktemp -d /tmp/test-kata-snp.XXXXXXXX)"
+  SSH_KEY_FILE="${TEST_DIR}/container-ssh-key"
+
+  # Install package dependencies
+  echo "Installing required packages..."
+  esudo apt install -y \
+    python-is-python3 \
+    python3-pip \
+    jq \
+    mysql-client \
+    docker-compose \
+    cpuid
+  pip install sev-snp-measure
+  "${TESTS_REPO_DIR}/.ci/install_yq.sh" >&2
```

Comment:
Not all of these are required currently. I don't think we use `cpuid` yet, for instance, since we don't calculate launch measurement. On the other hand maybe we should make these deps generic and move them to the common file.

Reply:
I like the idea of maintaining the packages to be installed in the script that needs them. I added these packages in anticipation for when attestation tests would be added, but for now I've trimmed them down to what's required.